puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/smart-proxy.groovy

```diff
@@ -0,0 +1,103 @@
+    }
+    environment {
+        BUNDLE_JOBS = 4
+        BUNDLE_RETRY = 3
+    }
+
+    stages {
+        stage('Rubocop') {
+            agent any
+            environment {
+                BUNDLE_WITHOUT = 'bmc:development:dhcp_isc_inotify:dhcp_isc_kqueue:journald:krb5:libvirt:puppetca_token_whitelisting:realm_freeipa:windows'
+            }
+
+            stages {
+                stage('Setup Git Repos') {
+                    steps {
+                        ghprb_git_checkout()
+                    }
+                }
+                stage('Setup RVM') {
+                    steps {
+                        configureRVM('2.7')
+                    }
+                }
+                stage('Install dependencies') {
+                    steps {
+                        withRVM(['bundle install'], '2.7')
+                    }
+                }
+                stage('Run Rubocop') {
+                    steps {
+                        withRVM(['bundle exec rubocop --format progress --out rubocop.log --format progress'], '2.7')
+                    }
+                    post {
+                        always {
+                            recordIssues tool: ruboCop(pattern: 'rubocop.log'), enabledForFailure: true
+                        }
+                    }
+                }
+            }
+            post {
+                always {
+                    cleanupRVM('2.7')
+                    deleteDir()
+                }
+            }
+        }
+
+        stage('Test') {
+            matrix {
+                agent any
+                axes {
+                    axis {
+                        name 'ruby'
+                        values '2.5', '2.6', '2.7'
+                    }
+                }
+                environment {
+                    BUNDLE_WITHOUT = 'development'
+                }
+                stages {
+                    stage('Setup Git Repos') {
+                        steps {
+                            ghprb_git_checkout()
+                        }
+                    }
+                    stage('Setup RVM') {
+                        steps {
+                            configureRVM(ruby)
+                        }
+                    }
+                    stage('Install dependencies') {
+                        steps {
+                            withRVM(['bundle install'], ruby)
+                        }
+                    }
+                    stage('Run Tests') {
+                        steps {
+                            withRVM(['bundle exec rake jenkins:unit'], ruby)
+                        }
+                        post {
+                            always {
+                                junit testResults: 'jenkins/reports/unit/*.xml'
+                            }
+                        }
+                    }
+                }
+                post {
+                    always {
+                        archiveArtifacts artifacts: 'Gemfile.lock'
```

Comment:
Perhaps if you move this to inside the matrix you could archive all of them, similar to how junit works? might need to rename them first to a different name for every version.

Reply:
This should already be part of the matrix. It probably needs to be renamed thought. However, I would like to get this merged first so I can continue with removing the checkstyle plugin.